geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreFactory.scala

```diff
@@ -235,3 +220,22 @@ object HBaseDataStoreFactory extends LazyLogging {
   }
 
 }
+
+// noinspection TypeAnnotation
+object HBaseDataStoreParams {
+  val HBaseCatalog             = new GeoMesaParam[String]("hbase.catalog", "Catalog table name", required = true, deprecated = Seq("bigtable.table.name"))
+  val ConnectionParam          = new GeoMesaParam[Connection]("hbase.connection", "Connection", deprecated = Seq("connection"))
+  val CoprocessorUrl           = new GeoMesaParam[Path]("hbase.coprocessor.url", "Coprocessor Url", deprecated = Seq("coprocessor.url"))
+  val RemoteFiltering          = new GeoMesaParam[java.lang.Boolean]("hbase.remote.filtering", "Remote filtering", default = true, deprecated = Seq("remote.filtering"))
+  val LooseBBox                = GeoMesaDataStoreFactory.LooseBBoxParam
+  val QueryThreads             = GeoMesaDataStoreFactory.QueryThreadsParam
+  val MaxRangesPerExtendedScan = new GeoMesaParam[java.lang.Integer]("hbase.ranges.max-per-extended-scan", "Max Ranges per Extended Scan", default = 100, deprecated = Seq("max.ranges.per.extended.scan"))
+  val GenerateStats            = GeoMesaDataStoreFactory.GenerateStatsParam
+  val AuditQueries             = GeoMesaDataStoreFactory.AuditQueriesParam
+  val QueryTimeout             = GeoMesaDataStoreFactory.QueryTimeoutParam
+  val Caching                  = GeoMesaDataStoreFactory.CachingParam
+  val EnableSecurity           = new GeoMesaParam[java.lang.Boolean]("hbase.security.enabled", "Enable HBase Security (Visibilities)", default = false, deprecated = Seq("security.enabled"))
+  val Authorizations           = org.locationtech.geomesa.security.AuthsParam
+  val ForceEmptyAuths          = org.locationtech.geomesa.security.ForceEmptyAuthsParam
+  val Namespace                = GeoMesaDataStoreFactory.NamespaceParam
```

Comment:
Did every DataStore get the namespace param?

Reply:
yeah this fixes a gml encoding bug

Reply:
actually it was part of a different PR, it's just continued here

Reply:
Since this param is necessary fix to common gotcha, I'd be tempted to roll it into a base trait for all our DataStoreFactory's.

Reply:
pulled into a trait